device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -127,14 +134,44 @@ else if (receivePeriod == 0) //default builder value for this option, signals th
      * <p>
      * If this client is already open, then this method will do nothing.
      * <p>
-     * @throws IOException If any IO errors occur while opening the multiplexed connection.
+     * @throws MultiplexingClientException If any IO or authentication errors occur while opening the multiplexed connection.
+     * @throws MultiplexingClientDeviceRegistrationFailedException If one or many of the registered devices failed to authenticate.
```

Comment:
Q - both of these exception types point towards authentication related errors, what is the difference in the scenarios where they are thrown?
MultiplexingClientException = while initial open vs MultiplexingClientDeviceRegistrationFailedException = registered but subsequently keys expired?

Reply:
```MultiplexingClientException``` is the super type of ```MultiplexingClientDeviceRegistrationFailedException``` here. Normally I'd only call out that the method throws the super type, since that is all that Javadocs require, but it felt useful to call out ```MultiplexingClientDeviceRegistrationFailedException``` in particular as it can be handled differently from ```MultiplexingClientException```. The more specific exception has a map of deviceId->registration exception that users can read when reacting to such exceptions.